src/libAtomVM/ets_hashtable.c

```diff
@@ -81,9 +81,49 @@ static void print_info(struct EtsHashTable *hash_table)
     }
 }
 #endif
+struct HNode *ets_hashtable_new_node(term entry, int keypos, GlobalContext *global)
+{
+
+    Heap *heap = malloc(sizeof(Heap));
+    if (IS_NULL_PTR(heap)) {
+        return NULL;
+    }
+    size_t size = (size_t) memory_estimate_usage(entry);
+    if (memory_init_heap(heap, size) != MEMORY_GC_OK) {
```

Comment:
I was wondering, why we create new heap instead of piggybacking on owner process' heap?

Reply:
Short answer: this code is just a refactored existing code block that was doing the same.

Long answer: the GC is completely decoupled from ETS at the moment and ETS tables are not tied at all to a Context, so GC will not use their items as roots, so if we use the process heap we would screw up them.

Reply:
If I understood correctly: they could be on the owner's heap if we could guarantee that they won't be GC'd.

Reply:
right, that would require some bigger changes and there might be some extra complexity.